packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -34,17 +31,50 @@ type BaseButtonCommonProps = {
   type?: 'button' | 'reset' | 'submit';
 };
 
+/*
+Mandatory children prop when icon is not provided
+*/
 type BaseButtonWithoutIconProps = BaseButtonCommonProps & {
   icon?: undefined;
   children: string;
 };
 
+/*
+ Optional children prop when icon is provided
+*/
 type BaseButtonWithIconProps = BaseButtonCommonProps & {
   icon: IconComponent;
   children?: string;
 };
 
-export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
+/*
+ With or without icon prop. We need at least an icon or a children prop present.
+*/
+type BaseButtonPropsWithOrWithoutIconProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
+
+/*
+ With a variant prop along with or without an icon prop.
+*/
+type BaseButtonWithVariantProps = BaseButtonPropsWithOrWithoutIconProps & {
+  variant?: 'primary' | 'secondary' | 'tertiary';
+  intent?: undefined;
+  contrast?: undefined;
+};
```

Comment:
not clear why do we write `intent?: undefined;`, can't we just ignore them? we can directly write this?
```suggestion
type BaseButtonWithVariantProps = BaseButtonPropsWithOrWithoutIconProps & {
  variant?: 'primary' | 'secondary' | 'tertiary';
};
```